core/amber/src/main/scala/edu/uci/ics/amber/engine/common/ambermessage/ControlMessage.scala

```diff
@@ -48,4 +49,6 @@ object ControlMessage {
   final case class ResumeTuple(faultedTuple: FaultedTuple) extends WorkflowMessage
 
   final case class KillAndRecover() extends WorkflowMessage
+
+  final case class LogErrorToFrontEnd(err: WorkflowRuntimeError) extends WorkflowMessage
```

Comment:
I suggest we change the name of this control message. I don't think it's good to involve frontend in the engine.